src/query/vacuum.c

```diff
@@ -7480,17 +7478,171 @@ vacuum_notify_need_flush (int need_flush)
+		     "VACUUM ERROR: Failed to obtain heap file identifier for class (%d, %d, %d)", class_oid.volid,
+		     class_oid.pageid, class_oid.slotid);
+
+      assert_release (false);
+      return error_code;
+    }
+
+  /* Overflow VFID is required to remove overflow pages. */
+  if (heap_ovf_find_vfid (thread_p, &helper.hfid, &helper.overflow_vfid, false, PGBUF_CONDITIONAL_LATCH) == NULL)
+    {
+      /* Failed conditional latch. Try again using unconditional latch. */
+      VACUUM_PERF_HEAP_TRACK_PREPARE (thread_p, &helper);
+
+      if (heap_ovf_find_vfid (thread_p, &helper.hfid, &helper.overflow_vfid, false, PGBUF_UNCONDITIONAL_LATCH) == NULL)
+	{
+	  assert_release (false);
+	  return ER_FAILED;
+	}
+    }
+  if (VFID_ISNULL (&helper.overflow_vfid))
+    {
+      assert_release (false);
+      return ER_FAILED;
+    }
+
+  /* Fix first overflow page (forward_page). */
+  helper.forward_page = pgbuf_fix (thread_p, &forward_vpid, OLD_PAGE, PGBUF_LATCH_WRITE, PGBUF_UNCONDITIONAL_LATCH);
+  if (helper.forward_page == NULL)
+    {
+      ASSERT_ERROR_AND_SET (error_code);
+      vacuum_er_log (VACUUM_ER_LOG_ERROR | VACUUM_ER_LOG_HEAP, "VACUUM ERROR: Failed to fix page (%d, %d).",
+		     forward_vpid.volid, forward_vpid.pageid);
+      return error_code;
+    }
+  else if (!pgbuf_check_page_ptype (thread_p, helper.forward_page, PAGE_OVERFLOW))
+    {
+      /* not overflow type of page found: possible if the job was interrupted */
+      assert (was_interrupted);
+      pgbuf_unfix_and_init (thread_p, helper.forward_page);
+      return NO_ERROR;
+    }
+
+  /* Read MVCC header from first overflow page. */
+  error_code = heap_get_mvcc_rec_header_from_overflow (helper.forward_page, &helper.mvcc_header, NULL);
+  if (error_code != NO_ERROR)
+    {
+      ASSERT_ERROR ();
+      vacuum_er_log (VACUUM_ER_LOG_ERROR | VACUUM_ER_LOG_HEAP,
+		     "VACUUM ERROR: Failed to get MVCC header from overflow page %d|%d.", forward_vpid.volid,
+		     forward_vpid.pageid);
+      pgbuf_unfix_and_init (thread_p, helper.forward_page);
+      return error_code;
+    }
+
+  /* Check if record can be vacuumed. */
+  helper.can_vacuum = mvcc_satisfies_vacuum (thread_p, &helper.mvcc_header, threshold_mvccid);
+  if (helper.can_vacuum == VACUUM_RECORD_REMOVE)
+    {
+      /* Record has been deleted and it can be removed. */
+
+      /* Unfix first overflow page. */
+      pgbuf_unfix_and_init (thread_p, helper.forward_page);
+
+      /* Encapsulate ovf delete into a system operation to ensure atomicity */
+      if (log_start_system_op (thread_p) == NULL)
+	{
+	  assert_release (false);
+	  return ER_FAILED;
+	}
+      if (heap_ovf_delete (thread_p, &helper.hfid, ovf_oid, &helper.overflow_vfid) == NULL)
+	{
+	  /* Failed to delete. */
+	  assert_release (false);
+	  log_end_system_op (thread_p, LOG_RESULT_TOPOP_ABORT);
+	  return ER_FAILED;
+	}
+
+      log_end_system_op (thread_p, LOG_RESULT_TOPOP_COMMIT);
+      VACUUM_PERF_HEAP_TRACK_LOGGING (thread_p, &helper);
```

Comment:
@andrei14vl 

Isn't it `VACUUM_PERF_HEAP_TRACK_EXECUTE`? It may be relocated to the end of the function. 

Please also examine other perf tracks of the function, for instance, `VACUUM_PERF_HEAP_TRACK_PREPARE` can be set on line 7604 (or 7592).